src/Microsoft.AspNet.JsonPatch/Adapters/ObjectAdapter.cs

```diff
@@ -638,9 +642,22 @@ public void Copy(Operation<T> operation, T objectToApplyTo)
 
         private bool IsNonStringArray(JsonPatchProperty patchProperty)
         {
+            var jsonPropertyType = patchProperty.Property.PropertyType.GetTypeInfo();
+
+            if (jsonPropertyType.IsGenericType)
+            {
+                var genericArguments = jsonPropertyType.GetGenericArguments();
+                if (genericArguments.Length != 1)
+                {
+                    return false;
+                }
+                var listType = typeof(IList<>).MakeGenericType(genericArguments);
+                return listType.GetTypeInfo().IsAssignableFrom(jsonPropertyType);
+            }
```

Comment:
The right check here is probably to look at `.ImplementedInterfaces` for a type which is is generic and has `IList<>` as it's generic definition. This code here isn't really accurate.